src/components/memorygame/cardInitialStateGenerator.jsx

```diff
@@ -0,0 +1,65 @@
+import * as invariant from "invariant";
+import CARD_BACKSIDE_IMAGES from "./memoryCardBackSideImages";
+import shuffle from "./shuffle";
+
+/**
+ * cardDataArr example:
+ * [{description:"...",link:"...",phrase:".."},{description:"...",link:"...",phrase:".."},...]
+ *
+ * CARD_BACKSIDE_IMAGES example:[image1,image2,...]
+ *
+ * Game example: https://www.brightfocus.org/alzheimers/memory-games/match-pictures
+ * Instead of squares this game uses cards.
+ *
+ * This function prepares the statesArr that sets the structure of game board
+ */
+const cardInitialStateGenerator = (cardDataArr, NUM_REQUIRED_MATCHES) => {
```

Comment:
fn parameters should never be ALL_CAPS, numRequiredMatches (it doesn't know whether or not its taking a constant, it doesnt care, teh function just knows it accepts a number)

Reply:
fixed